src/components/WalletSelectContent.js

```diff
@@ -48,6 +51,7 @@ const WalletSelectContent = ({}) => {
 
     let connector
     if (name === 'MetaMask') {
+      if (!active && chainId && !correctNetwork) await switchEthereumChain()
```

Comment:
call for coinbase wallet too? likely wallet connect too?

Reply:
also please add brackets - single statement `if` is bad form

Reply:
Walletconnect does not allow for switching in this way – walletconnect choice of chain is on the user's side.

I don't know coinbase wallet enough, but I believe it is the same.

Reply:
Should we call for all of the wallets and prompt on error?